Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/competitionprograms/Config.java

```diff
@@ -0,0 +1,201 @@
+package org.firstinspires.ftc.team417_CENTERSTAGE.competitionprograms;
+
+import com.acmerobotics.roadrunner.Pose2d;
+import com.qualcomm.robotcore.hardware.Gamepad;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Config {
+    // Put config state that's set by the menu here. Make these 'static' so that they can be read
+    // from both Auton and TeleOp:
+    public static boolean isRed = false;
+    public static boolean isClose = false;
+
+    public static boolean useOpenCV = true;
+    public static boolean doBackdropPixel = true;
+    public static boolean useDriveTo = true;
+    public static boolean useAprilTags = false;
+
+    public static String summary = "Config menu was not used :(";
+
+    public static Pose2d pose = new Pose2d(0, 0, 0);
+    public static int armPosition = 0;
+
+    static ArrayList<HashMap<Boolean, String>> configStrings;
+    static ArrayList<HashMap<Boolean, String>> summaryStrings;
+    static ArrayList<Boolean> defaults;
+
+    /**
+     * Initialize the menu.
+     */
+    static void init() {
+        configStrings = new ArrayList<>();
+
+        HashMap<Boolean, String> redStrings = new HashMap<>();
+        redStrings.put(true, "\uD83D\uDFE5 RED");
+        redStrings.put(false, "\uD83D\uDD35 BLUE");
+        configStrings.add(redStrings);
```

Comment:
Hashes are very resource intensive (require lots of CPU and memory to create). These could instead be simply 2-element arrays where the index is zero if false, one if true.

Reply:
@nobody-particular We should discuss different ways on how to refine this!